modules/Kconfig.open-amp

```diff
@@ -18,13 +19,11 @@ config OPENAMP_SRC_PATH
 
 config OPENAMP_MASTER
 	bool "OpenAMP Master Support"
-	default y
```

Comment:
The main reason of having OPENAMP_MASTER and OPENAMP_SLAVED enabled by default is that this is the default mode to compile OpenAMP library. Compiling the lib without at least one enable is not possible.
The purpose of these CONFIG is to decrease the footprint.

Reply:
How about changing it to choice?
1. Master + slave (default)
2. Master
3. Slave

It would prevent compiling without at least one and enable IPC Service to automatically decrease OpenAMP footprint, based on IPC Service configuration.

Reply:
Not sure to understand your proposal... For me this is the current implementation without your patch...
You mean having only one config enum instead of 2 boolean configs? 
Anyway i'm not the right person to give you feedback on the goodt practice here :-)

Reply:
Yes, I meant enum. With current implementation I cannot disable OPENAMP_SLAVE when IPC_SERVICE_MODE_MASTER is selected. If we replaced two bools in OpenAMP mode configuration with single enum, it could be configured automatically by IPC Service Kconfig.

Reply:
it is probably better to not do it. Some platforms need to use the library in both master and slave mode to communicate with several processors. I don't know if there is such platform is supported in zephyr project...
So perhaps they should not have dependency between IPC_SERVICE_MODE_xxx and OPENAMP_xxx configs. but as suggested in a previous comment to have a pre-build check.

Reply:
I understand. I'm reverting these changes